File: scripts/regenerate-lockfile.py
import subprocess
import os

project_dir = '/vercel/share/v0-project'
lock_file = os.path.join(project_dir, 'package-lock.json')

# Remove stale lock file if it exists
if os.path.exists(lock_file):
    os.remove(lock_file)
    print('Removed old package-lock.json')

# Generate fresh package-lock.json
print('Generating fresh package-lock.json...')
result = subprocess.run(
    ['npm', 'install', '--package-lock-only'],
    cwd=project_dir,
    capture_output=True,
    text=True,
    timeout=120
)

print('STDOUT:', result.stdout)
print('STDERR:', result.stderr)

if result.returncode == 0:
    print('Successfully generated package-lock.json')
else:
    print(f'Failed with return code: {result.returncode}')

File: scripts/regenerate-lockfile.mjs
import { execSync } from 'child_process';
import { existsSync, unlinkSync } from 'fs';
import { join } from 'path';

const projectDir = '/vercel/share/v0-project';

// Remove stale lock files if they exist
const packageLock = join(projectDir, 'package-lock.json');
if (existsSync(packageLock)) {
  unlinkSync(packageLock);
  console.log('Removed old package-lock.json');
}

// Generate fresh package-lock.json
console.log('Generating fresh package-lock.json...');
try {
  execSync('npm install --package-lock-only', {
    cwd: projectDir,
    stdio: 'inherit',
    timeout: 120000,
  });
  console.log('Successfully generated package-lock.json');
} catch (err) {
  console.error('Failed to generate lock file:', err.message);
  process.exit(1);
}
